modules/Layout/Header/Header.tsx

```diff
@@ -125,7 +125,7 @@ function Header({ hasError }: Props) {
                                     [styles.hidden]: isMenuOpen,
                                     [styles.close]: isSearchWidgetShown,
                                 })}
-                                icon={isSearchWidgetShown ? IconClose : IconSearch}
```

Comment:
❌ 

This was done for the mobile view, and there the icon does close the search. So there should be a more robust solution, in that on desktop it's always Search icon, but on mobile it switches depending on the state. You can use the `useDevice` hook to check if we're on mobile screen or not.

Reply:
I see, I missed the mobile view point here. I've changed it to isSearchWidgetShown && isMobile ? IconClose : IconSearch